Refactor release troubleshooting section and add another case (#10704)
DataDog/integrations-core: docs/developer/process/integration-release.md

        deleted tags, so any subsequent manual trigger tags will need to increment the version number.

    1. Delete the branch and tag, locally and on GitHub.
    
- If the [build pipeline](../meta/cd.md) failed due to another feature PR being merged after the release PR is opened, the merged release PR will not have updated and signed the feature PR's files.

>   

Reviewer: Aren't the 2 `Build pipeline failed` troubleshooting steps the same?

Author: the first one is sign only and the second is creating a manual version bump

Reviewer: Can we make it more explicit?

        deleted tags, so any subsequent manual trigger tags will need to increment the version number.

    1. Delete the branch and tag, locally and on GitHub.
    

> - If the [build pipeline](../meta/cd.md) failed due to another feature PR being merged after the release PR is opened, the merged release PR will not have updated and signed the feature PR's files.

Reviewer 2: ```suggestion
- If the [build pipeline](../meta/cd.md) failed due to another feature PR conflicting with and being merged after the release PR is opened, the merged release PR will not have updated and signed the feature PR's files.
```

> - If the [build pipeline](../meta/cd.md) failed due to another feature PR conflicting with and being merged after the release PR is opened, the merged release PR will not have updated and signed the feature PR's files.

Reviewer 2: What's the difference between this case and the one above? I know there's a difference, but it's unclear which one to use when.

Author: The first case is more common, where it's just a missing signature like PR was pushed but yubikey signing failed so only the changelog and integration versions were bumped

Reviewer 2: K, then maybe make that clear? 🙂 First, try this. If not, then try that...

  
    You may see an error like so:
    
    ```text
    in_toto.exceptions.RuleVerificationError: 'DISALLOW *' matched the following artifacts: ['/shared/integrations-core/datadog_checks_dev/datadog_checks/dev/tooling/commands/ci/setup.py']
    ```
      

>     1. Verify the hash signed in `.in-toto/tag.<KEYID>.link`, [(see example)](https://github.com/DataDog/integrations-core/blob/9836c71f15a0cb93c63c1d2950dcdc28b49479a7/.in-toto/tag.57ce2495.link), matches what's on master for the artifact in question.

Reviewer 2: ```suggestion
    1. Verify whether the hash signed in `.in-toto/tag.<KEYID>.link`, [(see example)](https://github.com/DataDog/integrations-core/blob/9836c71f15a0cb93c63c1d2950dcdc28b49479a7/.in-toto/tag.57ce2495.link) matches what's on `master` for the artifact in question.
```

>         To see the hash for the file, run the following `shasum` command (replace local file path):

Reviewer 2: ```suggestion
        To see the hash for the artifact, run the following `shasum` command (replace local file path):
```

        
        ```
        shasum -a 256 datadog_checks_dev/datadog_checks/dev/tooling/commands/ci/setup.py
        ```
    

>     1. Checkout and pull the most recent version of the `master` branch.

Reviewer 2: ```suggestion
    1. If any artifact mismatches, check out and pull the most recent version of the `master` branch.
```

    
        ```
        git checkout master
        git pull
        ```
            
    1. Release the integration again with a new version, bump the version appropriately.
    
        ```
        ddev release make <INTEGRATION> --version <VERSION>
        ```
    

>     1. Verify that the integration files are signed and update the integration changelog to reflect the feature PR title in the following format.

Reviewer 2: ```suggestion
    1. Verify that the integration files are signed, and update the integration changelog to reflect the feature PR title in the following format.
```

    in_toto.exceptions.RuleVerificationError: 'DISALLOW *' matched the following artifacts: ['/shared/integrations-core/datadog_checks_dev/datadog_checks/dev/tooling/commands/ci/setup.py']
    ```
      

>     1. Verify the signature signed in `.in-toto/tag.<hash>.link`, [(see example)](https://github.com/DataDog/integrations-core/blob/9836c71f15a0cb93c63c1d2950dcdc28b49479a7/.in-toto/tag.57ce2495.link), matches what's on master for the artifact in question.

Reviewer 2: ```suggestion
    1. Verify the hash signed in `.in-toto/tag.<KEYID>.link`, [(see example)](https://github.com/DataDog/integrations-core/blob/9836c71f15a0cb93c63c1d2950dcdc28b49479a7/.in-toto/tag.57ce2495.link), matches what's on master for the artifact in question.
```

>         To see the signature for the file, run the following `shasum` command (replace local file path):

Reviewer 2: ```suggestion
        To see the hash for the file, run the following `shasum` command (replace local file path):
```
